src/main/java/org/elasticsearch/index/mapper/internal/ParentFieldMapper.java

```diff
@@ -143,16 +151,18 @@ public ParentFieldMapper build(BuilderContext context) {
 
     private final String type;
     private final BytesRef typeAsBytes;
+    private final boolean docValuesJoin;
 
-    protected ParentFieldMapper(String name, String indexName, String type, @Nullable Settings fieldDataSettings, Settings indexSettings) {
+    protected ParentFieldMapper(String name, String indexName, String type, @Nullable Settings fieldDataSettings, Settings indexSettings, boolean docValuesJoin) {
```

Comment:
In fact, you can just pass the value of the 2.0 check to the super ctor for doc values? Then use hasDocValues() instead of having this `docValuesJoin`?